frontend/src/metabase/dashboard/actions/navigation.js

```diff
@@ -43,7 +44,9 @@ export const NAVIGATE_TO_NEW_CARD = "metabase/dashboard/NAVIGATE_TO_NEW_CARD";
 export const navigateToNewCardFromDashboard = createThunkAction(
   NAVIGATE_TO_NEW_CARD,
   ({ nextCard, previousCard, dashcard, objectId }) =>
-    (dispatch, getState) => {
+    async (dispatch, getState) => {
+      await dispatch(loadMetadataForDashcards([dashcard]));
```

Comment:
When the user is really quick and clicks on the card title before metadata is loaded, we need to wait here because everything below depends on metadata.